packages/vite/src/client/client.ts

```diff
@@ -400,46 +400,29 @@ async function fetchUpdate({ path, acceptedPath, timestamp }: Update) {
   const moduleMap = new Map<string, ModuleNamespace>()
   const isSelfUpdate = path === acceptedPath
 
-  // make sure we only import each dep once
-  const modulesToUpdate = new Set<string>()
-  if (isSelfUpdate) {
-    // self update - only update self
-    modulesToUpdate.add(path)
-  } else {
-    // dep update
-    for (const { deps } of mod.callbacks) {
-      deps.forEach((dep) => {
-        if (acceptedPath === dep) {
-          modulesToUpdate.add(dep)
-        }
-      })
-    }
-  }
```

Comment:
`modulesToUpdate` was always `new Set([])` or `new Set([acceptedPath])`.

Because:

- in `if (isSelfUpdate)` block
  - `path === acceptedPath` is `true` so it's calling `modulesToUpdate.add(acceptedPath)`
- in `else` block
  - it's only calling `modulesToUpdate.add(dep)` when `acceptedPath === dep`. So it's calling `modulesToUpdate.add(acceptedPath)`.